### PR TITLE
Removed the needs for system properties field

### PR DESCRIPTION
**Pull request by ppatierno**
### Type of change

- Refactoring

### Description

This PR does a little bit of refactoring by removing the need for the `javaSystemProperties` field in the `AbstractModel` and using the same approach for other JVM related options.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Why do you once use `ModelUtils.javaOptions` and once `jvmSystemProperties` with `ModelUtils.getJavaSystemPropertiesToString`? Wouldn't it make sense to unify it completely?

If not, should `jvmSystemProperties(...)` really ba part of the `AbstractModel`? I think this is one of the issues of the `AbstractModel` class. It is full of methods used in half of the model classes while other half does something completely different.

**Comment by ppatierno** [comment]
> Why do you once use ModelUtils.javaOptions and once jvmSystemProperties with ModelUtils.getJavaSystemPropertiesToString? Wouldn't it make sense to unify it completely?

I was expecting the question :-)
The `javaOptions` method is really a duplicate of others in the `AbstractModel` class and I am talking about `heapOptions`, `jvmPerformanceOptions` and `jvmSystemProperties`. One problem I saw it that the `heapOptions` seem to be really tight to Kafka (it sets the KAFKA_HEAP_OPTS env var) while `javaOptions` is more on STRIMZI_JAVA_OPTS. The `heapOptions` also has the heap fraction and max bytes parameters that configure other env vars not really used in other places (talking where `javaOptions` is used). I would agree that a better refactoring could be done, but I would also start with this step first.

> If not, should jvmSystemProperties(...) really ba part of the AbstractModel? I think this is one of the issues of the AbstractModel class. It is full of methods used in half of the model classes while other half does something completely different.

That's fine, I can move to the `ModeUtils` but then what about the others: `heapOptions` and `jvmPerformanceOptions`. I think they should be moved as well.

**Comment by scholzj** [comment]
> I was expecting the question :-)

Maybe having it in the comments would help if it is expected question :-)

> what about the others: heapOptions and jvmPerformanceOptions. I think they should be moved as well.

That is a bit hard to say for me without seeing how they are used etc. If the same applies to them as well then I guess yes.

**Comment by scholzj** [comment]
PS: i can see how removing the unnecessary field might be useful anyway even without doing some further movements. So if you want to stick with what you have in the PR now, its fine.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java [hunk 1]
```diff
@@ -1567,6 +1561,20 @@ protected void jvmPerformanceOptions(List<EnvVar> envVars) {
...
 
+    /**
+     * Adds STRIMZI_JAVA_SYSTEM_PROPERTIES variable to the EnvVar list if any system properties were specified.
+     *
+     * @param envVars List of Environment Variables to add to
+     */
+    protected void jvmSystemProperties(List<EnvVar> envVars) {
+        if (jvmOptions != null) {
+            String jvmSystemPropertiesString = ModelUtils.getJavaSystemPropertiesToString(jvmOptions.getJavaSystemProperties());
+            if (!jvmSystemPropertiesString.isEmpty()) {
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java:1572]: I think you need null check here as well.

**Comment by ppatierno** [comment]
@scholzj I moved all the methods we mentioned in the `ModelUtils` class which sounds better. After merging this PR I will take a look to unify with `javaOptions` method to avoid the code duplication we have today.

**Pull request merged**